engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/BusinessRuleTaskTest.java

```diff
@@ -334,4 +336,86 @@ public void shouldWriteDecisionEvaluationEvent() {
                       });
             });
   }
+
+  @Test
+  public void shouldWriteDecisionEvaluationEventIfEvaluationFailed() {
+    // given
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlClasspathResource(DMN_RESOURCE)
+            .withXmlResource(
+                processWithBusinessRuleTask(
+                    t ->
+                        t.zeebeCalledDecisionId("force_user").zeebeResultVariable(RESULT_VARIABLE)))
+            .deploy();
+
+    final var calledDecision =
+        deployment.getValue().getDecisionsMetadata().stream()
+            .filter(decision -> decision.getDecisionId().equals("force_user"))
+            .findFirst()
+            .orElseThrow();
```

Comment:
💭 This is probably something that we (and users potentially as well) will have to do more often. We could consider adding a convenience method for this if we encounter it more often.

Reply:
Sure. Let's keep an eye open.